cmd/clusterctl/pkg/client/repository/repository_local_unix.go

```diff
@@ -0,0 +1,240 @@
+		}
+	}
+	providerName := parts[len(parts)-3]
+	if providerName != providerConfig.Name() {
+		return nil, errors.Errorf("invalid path: path %q must contain provider name %q in the format {basepath}/{provider-name}/{version}/{components.yaml}", providerConfig.URL(), providerConfig.Name())
+	}
+	var basePath string
+	if len(parts) > 3 {
+		basePath = filepath.Join(parts[:len(parts)-3]...)
+	}
+	basePath = filepath.Clean("/" + basePath) // ensure basePath starts with "/"
+
+	repo := &localRepository{
+		providerConfig:        providerConfig,
+		configVariablesClient: configVariablesClient,
+		basepath:              basePath,
+		providerName:          providerName,
+		defaultVersion:        defaultVersion,
+		rootPath:              "", // Not applicable with localRepository
+		componentsPath:        componentsPath,
+	}
+
+	if defaultVersion == "latest" {
+		repo.defaultVersion, err = repo.getLatestRelease()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get latest version")
+		}
+	}
+
+	return repo, nil
+}
+
+// getLatestRelease returns the latest release for the local repository.
+func (r *localRepository) getLatestRelease() (string, error) {
+
+	// get all the sub-directories under {basepath}/{provider-name}/
+	releasesPath := filepath.Join(r.basepath, r.providerName)
+	files, err := ioutil.ReadDir(releasesPath)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to list release directories")
+	}
+	// search for the latest release according to semantic version
+	// releases with names that are not semantic version number are ignored
+	var latestTag string
+	var latestReleaseVersion *version.Version
+	for _, f := range files {
+		if f.IsDir() {
+			r := f.Name()
+			sv, err := version.ParseSemantic(r)
+			if err != nil {
+				// discard releases with tags that are not a valid semantic versions (the user can point explicitly to such releases)
+				continue
+			}
+			if sv.PreRelease() != "" || sv.BuildMetadata() != "" {
+				// discard pre-releases or build releases (the user can point explicitly to such releases)
+				continue
+			}
+			if latestReleaseVersion == nil || latestReleaseVersion.LessThan(sv) {
+				latestTag = r
+				latestReleaseVersion = sv
+			}
+		}
+	}
+
+	if latestTag == "" {
+		return "", errors.New("failed to find releases tagged with a valid semantic version number")
+	}
+
+	return latestTag, nil
+}
+
+// getVersions returns all the release versions for a local repository
+func (r *localRepository) getVersions() ([]string, error) {
+	// get all the sub-directories under {basepath}/{provider-name}/
+	releasesPath := filepath.Join(r.basepath, r.providerName)
+	files, err := ioutil.ReadDir(releasesPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list release directories")
+	}
+	var versions []string
+	for _, f := range files {
+		if f.IsDir() {
```

Comment:
what about 
```golang
if !f.IsDir() {
     continue
}
```
without nesting the code?

Reply:
Done.